pyomo/contrib/interior_point/interior_point.py

```diff
@@ -0,0 +1,669 @@
+                                                                     reallocation_factor=self.reallocation_factor,
+                                                                     max_iter=self.max_reallocation_iterations,
+                                                                     timer=timer)
+            if status not in {LinearSolverStatus.successful, LinearSolverStatus.singular}:
+                raise RuntimeError('Could not factorize KKT system; linear solver status: ' + str(status))
+
+            if status == LinearSolverStatus.successful:
+                neg_eig = self.linear_solver.get_inertia()[1]
+            else:
+                neg_eig = None
+            fact_con.log_info(_iter=self._iter, reg_iter=reg_iter, num_realloc=num_realloc,
+                              coef=0, neg_eig=neg_eig, status=status)
+            reg_iter += 1
+
+            if status == LinearSolverStatus.singular:
+                kkt = self.interface.regularize_equality_gradient(kkt=kkt,
+                                                                  coef=self.base_eq_reg_coef * self._barrier_parameter**0.25,
+                                                                  copy_kkt=False)
+
+            total_hess_reg_coef = self.hess_reg_coef
+            last_hess_reg_coef = 0
+
+            while neg_eig != desired_n_neg_evals or status == LinearSolverStatus.singular:
+                kkt = self.interface.regularize_hessian(kkt=kkt,
+                                                        coef=total_hess_reg_coef - last_hess_reg_coef,
+                                                        copy_kkt=False)
+                status, num_realloc = try_factorization_and_reallocation(kkt=kkt,
+                                                                         linear_solver=self.linear_solver,
+                                                                         reallocation_factor=self.reallocation_factor,
+                                                                         max_iter=self.max_reallocation_iterations,
+                                                                         timer=timer)
+                if status != LinearSolverStatus.successful:
+                    raise RuntimeError('Could not factorize KKT system; linear solver status: ' + str(status))
+                neg_eig = self.linear_solver.get_inertia()[1]
+                fact_con.log_info(_iter=self._iter, reg_iter=reg_iter, num_realloc=num_realloc,
+                                  coef=total_hess_reg_coef, neg_eig=neg_eig, status=status)
+                reg_iter += 1
+                if reg_iter > self.max_reg_iter:
+                    raise RuntimeError('Exceeded maximum number of regularization iterations.')
+                last_hess_reg_coef = total_hess_reg_coef
+                total_hess_reg_coef *= self.reg_factor_increase
+
+        return last_hess_reg_coef
+
+    def process_init(self, x, lb, ub):
```

Comment:
What do these methods do?

Reply:
They move variables to the interior of their bounds.